v1/remote/write.go

```diff
@@ -132,9 +136,13 @@ func (w *writer) initiateUpload(h v1.Hash) (location string, mounted bool, err e
 	uv := url.Values{
 		"mount": []string{h.String()},
 	}
+	l, err := w.img.LayerByDigest(h)
+	if err != nil {
+		return "", false, err
+	}
 	var from []string
-	for _, m := range w.options.MountPaths {
-		from = append(from, m.RepositoryStr())
+	if ml, ok := l.(*MountableLayer); ok {
+		from = append(from, ml.Repository.RepositoryStr())
```

Comment:
I don't think it makes a lot of sense for `from` to be an array, now.

Reply:
I combined this predicate with the one below.